packages/expo-updates/ios/EXUpdates/AppLoader/AppLoader.swift

```diff
@@ -202,35 +202,44 @@ open class AppLoader: NSObject {
       if let assets = updateManifest.assets(),
         !assets.isEmpty {
         self.assetsToLoad = assets
-        for asset in assets {
-          // before downloading, check to see if we already have this asset in the database
-          let matchingDbEntry = try? self.database.asset(withKey: asset.key)
-
-          if let matchingDbEntry = matchingDbEntry,
-            !matchingDbEntry.filename.isEmpty {
-            // merge fields from existing database entry into our current asset object
-            // retaining the original object since it's used in self->_assetsToLoad
-            // (this is different from on Android, where we keep the database-sourced object instead)
-            do {
-              try self.database.mergeAsset(asset, withExistingEntry: matchingDbEntry)
-            } catch {
-              self.logger.warn(message: "Failed to merge asset with existing database entry: \(error.localizedDescription)")
-            }
+        self.database.databaseQueue.async {
+          let embeddedUpdate = EmbeddedAppLoader.embeddedManifest(withConfig: self.config, database: self.database)
+          let extraHeaders = FileDownloader.extraHeadersForRemoteAssetRequest(
+            launchedUpdate: self.launchedUpdate,
+            embeddedUpdate: embeddedUpdate,
+            requestedUpdate: updateManifest
+          )
```

Comment:
```suggestion
        let embeddedUpdate = getEmbeddedUpdateOnDbQueue(...)
        let extraHeaders = FileDownloader.extraHeadersForRemoteAssetRequest(
          launchedUpdate: self.launchedUpdate,
          embeddedUpdate: embeddedUpdate,
          requestedUpdate: updateManifest
        )
        for asset in assets {
```

```
func getEmbeddedUpdateOnDbQueue(...) {
  return self.database.databaseQueue.sync {
    return EmbeddedAppLoader.embeddedManifest(withConfig: self.config, database: self.database)
  }
}
```

note that there's a slightly different behavior. the point here is not in databaseQueue but after your change it's inside databaseQueue. not sure if that could be a regression. maybe having a function with sync call would be safer

Reply:
Unfortunelly we can't really use `sync` in here, calling `self.database.databaseQueue.sync` inside the `startLoading` will always throw this error 
<img width="1029" alt="image" src="https://github.com/user-attachments/assets/a181e15e-bc2b-4021-aea2-2333fc0e9c8b" />

Reply:
interesting. thanks for sharing that. i suspected the crash from `self.database.databaseQueue.sync` is because it's already on the queue and it did.

it enters to the db queue already here https://github.com/expo/expo/blob/9e8ac382efcd243dcbd0f5d6fa104c8a7a2a7814/packages/expo-updates/ios/EXUpdates/AppLoader/AppLoader.swift#L146-L147
so you don't need to push async task to the db queue again:
https://github.com/expo/expo/blob/9e8ac382efcd243dcbd0f5d6fa104c8a7a2a7814/packages/expo-updates/ios/EXUpdates/AppLoader/AppLoader.swift#L205

Reply:
Ohh I see, I've just removed the second `databaseQueue` call